### PR TITLE
Change code

### PR DESCRIPTION
**Pull request by maflcko**
This changes the code.

PoS has several outstanding problems, see for example section 4.2
"Costless Simulation" of https://download.wpsoftware.net/bitcoin/pos.pdf .
Simply signing the blocks with a predetermined set of keys has similar
tradeoffs, but an often disregarded advantage: A limited set of (let's
say 2) block signers consumes 99.5% less energy than a larger set of
(let's say 400) participants in PoS.

Luckily, in Bitcoin Core this consensus mechanism is already implemented
and the code changes required are trivial. Nonetheless, review is
appreciated.

### CODE REVIEW

**Comment by hebasto** [comment]
Scientific ACK, CI failure is unrelated.

**Comment by laanwj** [comment]
Very changy PR, have confirmed that this changes the code, would recommend reviewing again 10/10

**Comment by jonatack** [comment]
Mistyped and git checked out this branch instead of PR 24729 but good to see some changes.

**Comment by fanquake** [comment]
I think 4 maintainers being present in this thread means we can just merge this, and tell every bitcoin user to immediately upgrade (shame we never integrated auto-updates). Great work everyone.

**Comment by laanwj** [comment]
@kallewoof might want to ACK this i'm not sure he's in to become the global financial signing authority, also he might want to make the signet faucet more robust first :smile:

**Comment by kallewoof** [comment]
Finally.

Strong concept ACK.

**Comment by fjahr** [comment]
Good initiative but I think we should also lower the minimum difficulty (nbits) as well. That is still a lot of wasted resources. Maybe there should instead be a maximum difficulty limit so that hashes with too much PoW will get rejected and mining with a faster machine is disincentivized.

**Comment by luke-jr** [comment]
Approach NACK. This is a hardfork without consensus.

But it should be trivial to reimplement as a softfork! All we have to do is add PoS on top of the current rules. Then miners will need their blocks signed off by stake (which could be their own or someone else's).

**Comment by bigspider** [comment]
Signatures are very computationally expensive to produce and validate.

We can save a further 99.5% of energy by replacing signatures with the string `"I want to send this transaction"`, which is also less than half the size of Schnorr signatures.

**Comment by jimbojw** [comment]
This is a good start, but storage uses energy too. Recommend piping block data to `/dev/null`. It’s fast as hell and supports sharding.

**Comment by brunoerg** [comment]
Strong concept ACK. CI failed but we could ignore and disable it to save energy.

**Comment by rajarshimaitra** [comment]
Strong Concept ACK..

Such change, much energy, so wow..

**Comment by 0xB10C** [comment]
While you're at it, please remove the low-r signature grinding from the wallet too. Better to waste block-space than energy. Otherwise LGTM.

**Comment by handsomelatino** [comment]
Idea is great, finally we can save the world and save energy. But honestly...

I am extremely disappointed at how little love was put into this PR. 11 out of 15 tests are failing, code like this should not be merged.

**Comment by jarolrod** [comment]
crACK

**Comment by renepickhardt** [comment]
> Simply signing the blocks with a predetermined set of keys has similar
> tradeoffs, but an often disregarded advantage: A limited set of (let's
> say 2) block signers consumes 99.5% less energy than a larger set of
> (let's say 400) participants in PoS.

I am a bit skeptical about your 99.5% estimate. I am sorry to say that it seems like a bit of overselling as there would still be the users who consume energy when verifying the signatures of the blocks and transactions. This should happen in your proposed solution as well as in a POS proposal and seems to dominate the savings resulting from your claimed advantage.

**Comment by Sjors** [comment]
> We can save a further 99.5% of energy by replacing signatures with the string `"I want to send this transaction"`, which is also less than half the size of a Schnorr signature.

utACK 746c75a33c3c822bf04e16d435a3fef7ccb7da4e modulo replace sha256 by md5 and make CSW the fiduciary

**Comment by jaonoctus** [comment]
LGTM, #ChangeTheCode

**Review by SergioGlorias — COMMENTED** [review]
How to Change Bitcoin

- [x] Make a change that is not for nothing and with problems

**Comment by sipa** [comment]
ACK #BIP325IsConsensus

**Comment by lucasdcf** [comment]
LGTM!

**Comment by Fittiboy** [comment]
Hey it's me, planet earth. LGTM, I think we should merge immediately! I'm already running the code and it's great, my oceans have stopped boiling entirely.

**Review by jaonoctus — APPROVED** [review]

**Review by murchandamus — APPROVED** [review]
ACK 746c75a33c3c822bf04e16d435a3fef7ccb7da4e I like the minimalist approach, easiest review ever. Thanks for changing the code.

**Comment by instagibbs** [comment]
NACK, signet needs to be forked to have the 20-35 people that control Bitcoin in the multisig to be secure

**Comment by Zero-1729** [comment]
crACK

It's about time, LGTM!

**Comment by Sjors** [comment]
@instagibbs why not 8 billion? Nothing a bit of MuSig2 and sustainably grown Merkle trees can't fix?

**Comment by katesalazar** [comment]
new site is not self hosted, sorry to NAK
xoxo

**Comment by studioTeaTwo** [comment]
What a beautiful PR! Minimal code changes. Simple Made Easy.

**Comment by luke-jr** [comment]
See [here](https://www.youtube.com/watch?v=oHg5SJYRHA0) for how to fix the CI tests

File: configure.ac
Comment (1440000bytes) [inline configure.ac:10]:
```suggestion
AC_INIT([Bitcoin ChangeTheCode],m4_join([.], _CLIENT_VERSION_MAJOR, _CLIENT_VERSION_MINOR, _CLIENT_VERSION_BUILD)m4_if(_CLIENT_VERSION_RC, [0], [], [rc]_CLIENT_VERSION_RC),[https://github.com/bitcoin/bitcoin/issues],[bitcoin],[https://nitter.net/cleanupbitcoin/])
```

**Comment by junderw** [comment]
> See [here](https://www.youtube.com/watch?v=oHg5SJYRHA0) for how to fix the CI tests

We're no strangers to code.

You know the compiler rules, and so do I.

**Comment by junderw** [comment]
tACK

I have verified this has decreased the energy usage of the Bitcoin network by 1 million fold.

Let's ship it as version 420.69.1337

**Comment by fanquake** [comment]
Nice to see so many new faces here, willing to review code. If you could redirect your efforts to any of the other 400 active code changes, that'd be great.

**Comment by laanwj** [comment]
>Good initiative but I think we should also lower the minimum difficulty (nbits) as well. 

> Hey it's me, planet earth. LGTM, I think we should merge immediately! I'm already running the code and it's great, my oceans have stopped boiling entirely.

unfortunately this is not sufficient to address the issue
even with these serious improvements, the silicon semiconductor supply chain is burdened with toxic externalities, gratuitous inefficiency as well as organizational problems
transitioning the computational means to fully biological and organic is blocked on some updates to the underlying system, these are planned for version v293.0
agree on closing this PR for now, thank you for supporting Change

**Pull request closed without merging**
